packages/jira-adapter/src/weak_references/field_configuration_items.ts

```diff
@@ -92,11 +102,8 @@ const removeMissingFields: WeakReferencesHandler['removeWeakReferences'] = ({ el
 
       const fixedInstance = instance.clone()
       fixedInstance.value.fields = Object.fromEntries(await awu(Object.entries(fieldConfigurationItems))
-        .filter(async ([_id, field]) =>
-          (isReferenceExpression(field.id)
-            // eslint-disable-next-line no-return-await
-            && await elementsSource.has(field.id.elemID)
-          )).toArray())
+        .filter(async ([fieldName, _field]) =>
+          fieldExists(fieldName, elementsSource)).toArray())
```

Comment:
no need to have `async` when there is no `await`
no need to specify additional arguments if they are no used...
```suggestion
        .filter(([fieldName]) => fieldExists(fieldName, elementsSource))
        .toArray())
```

also, this whole thing can be replaced with a much shorter and (IMO) more readable call to `pickAsync` from lowerdash

```typescript
fixedInstance.value.fields = await pickAsync(
  fieldConfigurationItems,
  (_field, fieldName) => fieldExists(fieldName),
)
```